### PR TITLE
Transformations - make substring and not substring transformer case insensitive.

### PR DESCRIPTION
**Pull request by timlevett**
**What is this feature?**

Got some [feedback](https://raintank-corp.slack.com/archives/C036J5B39/p1709890324917879?thread_ts=1709852108.538739&cid=C036J5B39) on the substring transformer for value filtering that it would be nicer if it was case insensitive. So this adds that :)

**Why do we need this feature?**

Sometimes you don't know if something is cased when searching for it.

**Who is this feature for?**

everyone

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by baldm0mma — APPROVED** [review]

File: packages/grafana-data/src/transformations/matchers/valueMatchers/substringMatchers.ts [hunk 1]
```diff
@@ -28,7 +30,13 @@ const isNotSubstringValueMatcher: ValueMatcherInfo<BasicValueMatcherOptions> = {
   get: (options) => {
     return (valueIndex: number, field: Field) => {
       const value = field.values[valueIndex];
-      return typeof value === 'string' && options.value !== '' && !value.includes(options.value);
+      return (
+        typeof value === 'string' &&
+        options.value &&
+        value &&
```
Comment (leeoniya) [inline packages/grafana-data/src/transformations/matchers/valueMatchers/substringMatchers.ts:36]: is this the `0` coercion problem back?

**Comment by leeoniya** [comment]
imo this should be an option, not the only behavior. fine to make it default tho.

File: packages/grafana-data/src/transformations/matchers/valueMatchers/substringMatchers.ts (hunk 1 above)
Comment (leeoniya) [inline packages/grafana-data/src/transformations/matchers/valueMatchers/substringMatchers.ts:36]: is this the `'0'` coercion problem back?

**Review by codeincarnate — APPROVED** [review]
Looks good!

**Comment by timlevett** [comment]
>imo this should be an option, not the only behavior. fine to make it default tho.

I agree @leeoniya, but I didn't see an easy way to introduce a checkbox to the filter by value model, would take more effort. What do you think, should we hold off on this but then file a ticket to do it properly?

File: packages/grafana-data/src/transformations/matchers/valueMatchers/substringMatchers.ts (hunk 1 above)
Comment (timlevett) [reply packages/grafana-data/src/transformations/matchers/valueMatchers/substringMatchers.ts:36]: possibly, let me check

**Comment by nmarrs** [comment]
> > imo this should be an option, not the only behavior. fine to make it default tho.
> 
> I agree @leeoniya, but I didn't see an easy way to introduce a checkbox to the filter by value model, would take more effort. What do you think, should we hold off on this but then file a ticket to do it properly?

IMO I think we can move forward with merging this as having the default behavior be case insensitive makes sense to me as it is the least limiting option. In future based on usage / feedback we can consider adding more advanced filtering / matching modes to this

**Review by nmarrs — APPROVED** [review]

**Comment by leeoniya** [comment]
:+1: all good, as long as the 0-coercion thing is not an issue.

**Comment by timlevett** [comment]
Tested with 0 and its not an issue.

**Pull request merged**